### PR TITLE
refactor: Table expandable item props internal api rework

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Updated internal table APIs to support upcoming change https://github.com/cloudscape-design/components/pull/2108.

### How has this been tested?

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/table/internal.tsx [hunk 1]
```diff
@@ -532,9 +534,7 @@ const InternalTable = React.forwardRef(
                                   submitEdit={cellEditing.submitEdit}
                                   columnId={column.id ?? colIndex}
                                   colIndex={colIndex + colIndexOffset}
-                                  // Expandable props only apply to the first data column of the table.
-                                  // When present, the cell content is decorated with expand toggles and extra paddings.
-                                  expandableProps={isExpandable && colIndex === 0 ? expandableProps : undefined}
+                                  {...expandableCellProps}
```
Comment (pan-kot) [inline src/table/internal.tsx:537]: This change it necessary to not provide empty onExpandableItemToggle when only offset is needed, when a cell does not have the expand toggle but requires an offset.

File: src/table/body-cell/index.tsx [hunk 2]
```diff
@@ -157,10 +156,12 @@ export function TableBodyCell<ItemType>({
 }: TableBodyCellProps<ItemType> & { isEditable: boolean }) {
   const editDisabledReason = rest.column.editConfig?.disabledReason?.(rest.item);
 
-  if (editDisabledReason && !rest.expandableProps) {
+  // Inline editing is deactivated for expandable column because editable cells are interactive
```
Comment (Al-Dani) [inline src/table/body-cell/index.tsx:159]: should we put this in usage guidelines too? I know they're pretty complex already, but i can imagine this case to come up one day

File: src/table/body-cell/index.tsx [hunk 3]
```diff
@@ -157,10 +156,12 @@ export function TableBodyCell<ItemType>({
...
+  // and cannot include interactive content such as expand toggles.
+  if (editDisabledReason && rest.level === undefined) {
```
Comment (Al-Dani) [inline src/table/body-cell/index.tsx:161]: I'd suggest to create a constant with helpful name for the future
Smth like
```js
const cellWithExpandableToggle = rest.level !== undefined
if(editDisabledReason && !cellWithExpandableToggle) {..}
```

File: src/table/body-cell/index.tsx (hunk 2 above)
Comment (pan-kot) [reply src/table/body-cell/index.tsx:159]: We plan to remove this constraint in the near future. It wasn't done as part of the initial release due to design and dev complexity.

**Review by Al-Dani — APPROVED** [review]

**Pull request merged**